stl/inc/atomic

```diff
@@ -441,40 +493,41 @@ struct _Atomic_storage {
     // Provides operations common to all specializations of std::atomic, load, store, exchange, and CAS.
     // Locking version used when hardware has no atomic operations for sizeof(_Ty).
 
+    using _TVal  = remove_reference_t<_Ty>;
+    using _Guard = _Atomic_lock_guard<typename _Atomic_storage_types<_Ty>::_Spinlock>;
+
     _Atomic_storage() = default;
 
-    /* implicit */ constexpr _Atomic_storage(const _Ty _Value) noexcept : _Storage(_Value) {
+    /* implicit */ constexpr _Atomic_storage(conditional_t<is_reference_v<_Ty>, _Ty&, const _TVal> _Value) noexcept

```

Comment:
Assuming that this never has to handle rvalue references, `conditional_t<is_reference_v<_Ty>, _Ty&, const _TVal>` seems redundant and could be `conditional_t<is_reference_v<_Ty>, _Ty, const _TVal>`. Occurs below. If this is intentional then I suppose it is fine to keep.

Reply:
Yes, it was redundant, removed